### PR TITLE
Add new lidiag PVs for modulators

### PR DESCRIPTION
**Pull request by anacso17**

### CODE REVIEW

File: siriuspy/siriuspy/diagsys/lidiag/pvs.py [hunk 1]
```diff
@@ -134,6 +134,178 @@ def compute_update(self, computed_pv, updated_pv_name, value):
...
+            not computed_pv.pvs[LIPUStatusPV.PV_LVRDYOK].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_HVRDYOK].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_TRRDYOK].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_SELFFLT].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_SYS_RDY].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_TRGNORM].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_PLSCURR].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_VLTDIFF].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_CRRDIFF].connected
+        if disconnected:
```
Comment (fernandohds564) [inline siriuspy/siriuspy/diagsys/lidiag/pvs.py:200]: actually here you just neet to set all bits to 1, right? you could just do: `value = 2**(NBITS+1)-1`

File: siriuspy/siriuspy/diagsys/lidiag/pvs.py [hunk 2]
```diff
@@ -134,6 +134,178 @@ def compute_update(self, computed_pv, updated_pv_name, value):
...
+            value |= LIPUStatusPV.BIT_TRGNORM
+            value |= LIPUStatusPV.BIT_PLSCURR
+            value |= LIPUStatusPV.BIT_VLTDIFF
+            value |= LIPUStatusPV.BIT_CRRDIFF
+            return {'value': value}
+
+        # Run/Stop
+        sts = computed_pv.pvs[LIPUStatusPV.PV_RUNSTOP].value
+        if sts != 1 or sts is None:
+            value |= LIPUStatusPV.BIT_RUNSTOP
```
Comment (fernandohds564) [inline siriuspy/siriuspy/diagsys/lidiag/pvs.py:224]: if the BIT constant were the number of the bit and not the value, you would be able to use our method `update_bit` like this:
`update_bit(value, LIPUStatusPV.BIT_RUNSTOP, <sts_logic>)`

File: siriuspy/siriuspy/diagsys/lidiag/pvs.py [hunk 3]
```diff
@@ -134,6 +134,178 @@ def compute_update(self, computed_pv, updated_pv_name, value):
...
+    BIT_HVRDYOK = 0b000000010000000000
+    BIT_TRRDYOK = 0b000000100000000000
+    BIT_SELFFLT = 0b000001000000000000
+    BIT_SYS_RDY = 0b000010000000000000
+    BIT_TRGNORM = 0b000100000000000000
+    BIT_PLSCURR = 0b001000000000000000
+    BIT_VLTDIFF = 0b010000000000000000
+    BIT_CRRDIFF = 0b100000000000000000
+
+    PV_RUNSTOP = 0
```
[the remark this answers is not in the document]
Comment (anacso17) [reply siriuspy/siriuspy/diagsys/lidiag/pvs.py]: done!

File: siriuspy/siriuspy/diagsys/lidiag/pvs.py [hunk 4]
```diff
@@ -134,6 +134,178 @@ def compute_update(self, computed_pv, updated_pv_name, value):
...
+    PV_VLTDIFF = 15
+    PV_CRRDIFF = 16
+
+    def compute_update(self, computed_pv, updated_pv_name, value):
+        """Compute Status PV."""
+        value = 0
+
+        # connected?
+        disconnected = \
+            not computed_pv.pvs[LIPUStatusPV.PV_RUNSTOP].connected or \
```
[the remark this answers is not in the document]
Comment (anacso17) [reply siriuspy/siriuspy/diagsys/lidiag/pvs.py]: done!

File: siriuspy/siriuspy/diagsys/lidiag/pvs.py [hunk 5]
```diff
@@ -134,6 +134,178 @@ def compute_update(self, computed_pv, updated_pv_name, value):
         return {'value': value}
 
 
+class LIPUStatusPV:
+    """LI PU Status PV."""
+
+    BIT_DISCONN = 0b000000000000000001
```
[the remark this answers is not in the document]
Comment (anacso17) [reply siriuspy/siriuspy/diagsys/lidiag/pvs.py]: done!

File: siriuspy/siriuspy/diagsys/lidiag/pvs.py [hunk 6]
```diff
@@ -134,6 +134,178 @@ def compute_update(self, computed_pv, updated_pv_name, value):
...
+            not computed_pv.pvs[LIPUStatusPV.PV_HVRDYOK].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_TRRDYOK].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_SELFFLT].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_SYS_RDY].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_TRGNORM].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_PLSCURR].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_VLTDIFF].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_CRRDIFF].connected
+        if disconnected:
+            value |= LIPUStatusPV.BIT_DISCONN
```
[the remark this answers is not in the document]
Comment (anacso17) [reply siriuspy/siriuspy/diagsys/lidiag/pvs.py]: done!

File: siriuspy/siriuspy/diagsys/lidiag/pvs.py (hunk 2 above)
Comment (anacso17) [reply siriuspy/siriuspy/diagsys/lidiag/pvs.py]: done!

**Comment by anacso17** [comment]
> looping here is not an option? you could define a list with the fields with same logic for the `sts` and loop over this list.
> It seems two loops would solve it, wouldn't it? One for the logic `sts !=1 or sts is None` and other for `severity !=0`

much better!

**Review by fernandohds564 — APPROVED** [review]

**Review by xresende — APPROVED** [review]

**Pull request merged**